bark/models/behavior/safety_behavior/safety_behavior.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#include <tuple>
+#include <memory>
+
+#include "bark/models/behavior/safety_behavior/safety_behavior.hpp"
+#include "bark/models/behavior/idm/idm_lane_tracking.hpp"
+#include "bark/world/observed_world.hpp"
+
+namespace bark {
+namespace models {
+namespace behavior {
+
+Trajectory BehaviorSafety::Plan(
+    float min_planning_time, const world::ObservedWorld& observed_world) {
+  SetBehaviorStatus(BehaviorStatus::VALID);
+
+  // for now we support only the BehaviorIDMLaneTracking
+  BARK_EXPECT_TRUE(std::dynamic_pointer_cast<BehaviorIDMLaneTracking>(behavior_model_));
+
+  std::shared_ptr<BehaviorIDMLaneTracking> idm_lane_tracking_behavior =
```

Comment:
If you do a cast here without invalidity check, you should change the member pointer directly to the required type. Why do we need a pointer at all? Why not just derive from BehaviorIDMLaneTracking? In the constructor we already overwrite specific parameters for BehaviorIDMLaneTracking.

Reply:
well, in case we want to use another one in the future? e.g. MCTS safety behavior or motion primitives or..

Reply:
Okay, but how you implemented the safety behavior at the moment, it always requires the function SetConstantLaneCorridor(), when we use an MCTS we will rather replace the safety maneuver with the MCTS in the RSSConformantBehaviorModel instead of using an MCTS here.

Reply:
Isnt this discussion a bit academic? :)

Reply:
Actually it makes things more complicated for serialization than they have to be a.t.m. Thus, for me its not academic but introducing more work :-)

Reply:
For now, I de/serialized the pointer individually, but I see your point.